components/toast_wrapper/index.jsx

```diff
@@ -10,19 +10,40 @@ import {Posts} from 'mattermost-redux/constants';
 import {getAllPosts, getPostIdsInChannel} from 'mattermost-redux/selectors/entities/posts';
 import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
 import {makePreparePostIdsForPostList} from 'mattermost-redux/utils/post_list';
-import {countCurrentChannelUnreadMessages, isManuallyUnread} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentChannel, countCurrentChannelUnreadMessages, isManuallyUnread} from 'mattermost-redux/selectors/entities/channels';
+import {isCollapsedThreadsEnabled} from 'mattermost-redux/selectors/entities/preferences';
 
 import {updateToastStatus} from 'actions/views/channel';
 
 import ToastWrapper from './toast_wrapper.jsx';
+export function makeGetRootPosts() {
+    return createSelector(
+        getAllPosts,
+        getCurrentUserId,
+        getCurrentChannel,
+        (allPosts, currentUserId, channel) => {
+            // Count the number of new posts made by other users that haven't been deleted and are root posts
+            return Object.values(allPosts).filter((post) => {
+                return post.root_id === '' &&
+                    post.channel_id === channel.id &&
+                    post.user_id !== currentUserId &&
+                    post.state !== Posts.POST_DELETED;
+            }).reduce((map, obj) => {
+                map[obj.id] = true;
+                return map;
+            }, {});
+        },
+    );
+}
 
 export function makeCountUnreadsBelow() {
     return createSelector(
         getAllPosts,
         getCurrentUserId,
+        isCollapsedThreadsEnabled,
         (state, postIds) => postIds,
         (state, postIds, lastViewedBottom) => lastViewedBottom,
-        (allPosts, currentUserId, postIds, lastViewedBottom) => {
+        (allPosts, currentUserId, postIds, lastViewedBottom, isCollapsed) => {
```

Comment:
The arguments to the final result function are out of order

Reply:
damn! nice catch @ashishbhate